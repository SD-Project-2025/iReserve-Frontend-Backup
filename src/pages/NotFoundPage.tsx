"use client"

import { Box, Typography, Button } from "@mui/material"
import { useNavigate } from "react-router-dom"

const NotFoundPage = () => {
  const navigate = useNavigate()

  return (
    <Box
      component="section"
      sx={{
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        justifyContent: "center",
        minHeight: "calc(100vh - 200px)",
        textAlign: "center",
        p: 3,
      }}
    >
      <Typography variant="h1" component="h1" gutterBottom>
        404
      </Typography>
      <Typography variant="h4" component="h2" gutterBottom>
        Page Not Found
      </Typography>
      <Typography variant="body1" paragraph>
        The page you are looking for doesn't exist or has been moved.
      </Typography>
      <Button variant="contained" onClick={() => navigate("/dashboard")} sx={{ mt: 2 }}>
        Go to Dashboard
      </Button>
    </Box>
  )
}

export default NotFoundPage
