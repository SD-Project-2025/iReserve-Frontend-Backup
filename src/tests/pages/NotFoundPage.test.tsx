//@ts-ignore
import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import { useNavigate } from 'react-router-dom'
import NotFoundPage from '../../pages/NotFoundPage'

import '@testing-library/jest-dom'

// Mock the useNavigate hook
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}))

describe('NotFoundPage Component', () => {
  const mockNavigate = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
    ;(useNavigate as jest.Mock).mockReturnValue(mockNavigate)
  })

  it('renders correctly with all elements', () => {
    render(<NotFoundPage />)
    
    expect(screen.getByText('404')).toBeInTheDocument()
    expect(screen.getByText('Page Not Found')).toBeInTheDocument()
    expect(screen.getByText(/The page you are looking for doesn't exist or has been moved./i)).toBeInTheDocument()
    expect(screen.getByText('Go to Dashboard')).toBeInTheDocument()
  })

  it('navigates to dashboard when button is clicked', () => {
    render(<NotFoundPage />)
    
    const dashboardButton = screen.getByText('Go to Dashboard')
    fireEvent.click(dashboardButton)
    
    expect(mockNavigate).toHaveBeenCalledTimes(1)
    expect(mockNavigate).toHaveBeenCalledWith('/dashboard')
  })

  it('applies correct styles to the container', () => {
    render(<NotFoundPage />)
    
    const container = screen.getByText('404').parentElement
    expect(container).toHaveStyle('display: flex')
    expect(container).toHaveStyle('flex-direction: column')
    expect(container).toHaveStyle('align-items: center')
    expect(container).toHaveStyle('justify-content: center')
    expect(container).toHaveStyle('min-height: calc(100vh - 200px)')
    expect(container).toHaveStyle('text-align: center')
    expect(container).toHaveStyle('padding: 24px') // p: 3 translates to 24px
  })

  it('applies correct styles to the button', () => {
    render(<NotFoundPage />)
    
    const button = screen.getByText('Go to Dashboard')
    expect(button).toHaveStyle('margin-top: 16px') // mt: 2 translates to 16px
  })

  it('has proper heading hierarchy', () => {
    render(<NotFoundPage />)
    
    const h1 = screen.getByRole('heading', { level: 1 })
    expect(h1).toHaveTextContent('404')
    
    const h2 = screen.getByRole('heading', { level: 2 })
    expect(h2).toHaveTextContent('Page Not Found')
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<NotFoundPage />)
    expect(asFragment()).toMatchSnapshot()
  })
})