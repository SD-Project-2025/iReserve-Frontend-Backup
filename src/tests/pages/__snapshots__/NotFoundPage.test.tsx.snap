// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage Component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="MuiBox-root css-ij0kac"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom css-a4d0r7-MuiTypography-root"
    >
      404
    </h1>
    <h2
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
    >
      Page Not Found
    </h2>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-z2eky3-MuiTypography-root"
    >
      The page you are looking for doesn't exist or has been moved.
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-lll1vm-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Go to Dashboard
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </section>
</DocumentFragment>
`;
